GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerFixture/GVFSUpgradeReminderTests.cs

```diff
@@ -62,6 +63,56 @@ public void NoReminderForLeftOverDownloads()
             this.VerifyUpgradeVerbStopsReminder();
         }
 
+        [TestCase]
+        public void UpgradeTimerScheduledOnServiceStart()
+        {
+            this.RestartService();
+
+            bool timerScheduled = false;
+            for (int trialCount = 0; trialCount < 15; trialCount++)
+            {
+                Thread.Sleep(TimeSpan.FromSeconds(1));
+                if (this.ServiceLogContainsUpgradeMessaging())
+                {
+                    timerScheduled = true;
+                    break;
+                }
+            }
+
+            timerScheduled.ShouldBeTrue();
+        }
+
+        private bool ServiceLogContainsUpgradeMessaging()
+        {
+            string upgradeTimerMessage = "Checking for product upgrades. (Start)";

```

Comment:
maybe call this `expectedTimerMessage`, and include a comment about where it comes from?